addons/test_mail/models/test_mail_models.py

```diff
@@ -89,9 +89,7 @@ def _track_template(self, changes):
         return res
 
     def _creation_subtype(self):
-        if self.umbrella_id:
-            return self.env.ref('test_mail.st_mail_test_full_umbrella_upd')
-        return super(MailTestFull, self)._creation_subtype()
+        return self.env.ref('test_mail.st_mail_test_full_umbrella_upd')
```

Comment:
Why this change ?

Reply:
Because `_creation_subtype` was made `@api.model` in order to extract it from the iteration and allow to call `_message_log_batch`.

The only place where it wasn't only model dependent was on this test.

Reply:
Then probably your code is invalid, that's the whole purpose of that kind of tests.

Subtype creation could be dependent of your creation values. Anyway _message_log_batch should be called if no subtype is given back, as otherwise it has to be done per record.

Reply:
It's convenient to drop that support since it is not used in practice.

To keep the support for this, would mean to pre-group the threads per subtype, which I wanted to avoid for simplicity of code.

But if you think it should absolutely be kept even though not used, it's fine by me, I just need to know.

Reply:
Well it does not seem costly to group threads by subtype. I would prefer keeping the feature and left test model untouched to ease cross version comparisons of mail...

Reply:
Adapted code to keep behavior, now doing threads without subtype in batch, after the loop.